### PR TITLE
[DLP] Implemented dlp_deidentify_time_extract sample

### PR DESCRIPTION
**Pull request by Avani-Thakker-Crest**
## Description
Implemented dlp_deidentify_time_extract sample. Reference: https://cloud.google.com/dlp/docs/transformations-reference#time-extract

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/snippets/deid.py [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 import argparse
 
+from typing import List
```
Comment (micheldavid) [inline dlp/snippets/deid.py:21]: duplicated inside the added dlp_deidentify_time_extract as well. remove this one?

File: dlp/snippets/deid.py [hunk 2]
```diff
@@ -915,6 +917,135 @@ def write_data(data: types.storage.Value) -> str:
...
+
+
+def deidentify_with_time_extract(
+    project: str,
+    date_fields: List[str],
+    input_csv_file: str,
+    output_csv_file: str,
+) -> None:
+    """ Uses the Data Loss Prevention API to extract time part
+        data from a table.
```
Comment (micheldavid) [inline dlp/snippets/deid.py:935]: from a CSV file*

the underlying api uses a table, yes, but i think mentioning table here where there is no table object being passed into this function makes it harder to understand. you could mention that the CSV file is converted to a table input for DLP, and after deidentification, it is re-converted back into a CSV output file.

File: dlp/snippets/deid.py [hunk 3]
```diff
@@ -915,6 +917,135 @@ def write_data(data: types.storage.Value) -> str:
...
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        date_fields: A list of (date) fields in CSV file to de-identify
```
Comment (micheldavid) [inline dlp/snippets/deid.py:938]: mention the expected format for values in these columns: mm/DD/YYYY

File: dlp/snippets/deid.py [hunk 4]
```diff
@@ -915,6 +917,135 @@ def write_data(data: types.storage.Value) -> str:
...
+    # Convert date field list to Protobuf type.
+    def map_fields(field):
+        return {"name": field}
+
+    if date_fields:
+        date_fields = map(map_fields, date_fields)
+    else:
+        date_fields = []
+
+    f = []
```
Comment (micheldavid) [inline dlp/snippets/deid.py:958]: took me a while to figure out that this variable holds a list of cells from the csv file. each list of cells corresponds to one line from the csv file.
consider renaming it to make it easier to understand. suggestions: csv_lines, csv_cells_list

**Review by micheldavid — CHANGES_REQUESTED** [review]

File: dlp/snippets/deid.py (hunk 1 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: yes, updated.

File: dlp/snippets/deid.py (hunk 2 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: Thank you for noticing that out. I have made required changes, please check.

File: dlp/snippets/deid.py (hunk 4 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: Updated.

**Review by micheldavid — APPROVED** [review]

**Review by rsamborski — APPROVED** [review]

**Pull request merged**